main.c

```diff
@@ -0,0 +1,721 @@
+                inputLastName(lastName);
+                
+                dataOfStudent(firstName, lastName, school);
+                break;
+            }
+            case Showall:
+                printAllStudents(school);
+                break;
+            case Top10:{
+                printf("\n\tPlease Enter a Course (0-9): ");
+                int course ;
+                if (scanf("%d", &course) == 1) {
+                    struct TopTenStudents topTenStudent ;
+                    tenBestStudentsOfLayers(school, course, &topTenStudent);
+                    break;
+                }
+                }
+            case UnderperformedStudents:
+                underperformedSstudents(school);
+                break;
+            case Average:{
+                int course;
+                int layer = inputLayer();
+                
+                printf("\n\tPlease Enter a Course (1-10): ");
+                if (scanf("%d", &course) == 1) {
+                    if (course < 1 || course > 10) {
+                        printf("\n\t A wrong course (1-10): ");
+                        break;
+                    }
+                }
+                
+                long average = averageGradesOfCourse(layer, course, school);
+                printf("An average of layer %d of course %d is %ld: ", layer, course, average);
+                break;
+            }
+            case Export:
+                writeAllStudentsToFile(school, "/Users/alex/Documents/ios_projects/check_point_project/check_point_project/students_data.txt");
+                break;
+            case Exit:
+                break;
+            default:
+                printf("\nThere is no item with symbol \"%c\".Please enter a number between 1-10!\nPress any key to continue...",
+                       input);
+                getc(stdin);
+                getc(stdin);
+                break;
+        }
+    } while (input != Exit);
+}
+
+
+int main() {
+    // Initialize the data structures
+    struct School school;
+    memset(&school, 0, sizeof(struct School));
+
+    // Open the file for reading
+    FILE* file = fopen("/Users/alex/Documents/ios_projects/check_point_project/check_point_project/students-with-class.txt", "r");
+    if (file == NULL) {
+        printf("Error opening file.\n");
+        return 1;
+    }
+
+    // Read data from the file and populate the data structures
+    char buffer[200];
+    while (fgets(buffer, sizeof(buffer), file) != NULL) {
+        char firstName[50];
+        char lastName[50];
+        char phoneNumber[20];
+        int gradeNumber;
+        int classNumber;
+        int grades[10];
+
+        sscanf(buffer, "%s %s %s %d %d %d %d %d %d %d %d %d %d %d %d",
+               firstName, lastName, phoneNumber,
+               &gradeNumber, &classNumber,
+               &grades[0], &grades[1], &grades[2], &grades[3], &grades[4],
+               &grades[5], &grades[6], &grades[7], &grades[8], &grades[9]);
```

Comment:
What if you had an array with a size of 100?
prefer using a loop here for the grades.

Reply:
Fixed it.